### ISSUE TITLE
MAC M1 with Funcube Pro+

### ISSUE THREAD

**Issue by UncleBuck62** [f4exb/sdrangel#1909]
I am just getting back into SDR after a long hiatus.

This is my set up...
MAC M1 Studio running Sonoma
Funcube Dongle Pro+

I am testing it with an AM Demod picking up a local radio station at 1422 KHz

SDRangel sees the Funcube but it will not run at a sample rate of anything above 96000.
When running at 96000, the audio comes through but it is broken (pauses about every ¼ to ⅛ second) and what I do get sounds like chipmunks.

Have I missed something in the setup?

**Comment by srcejon** [comment]
Can you post a screenshot?

**Comment by UncleBuck62** [comment]
Not sure what you want the screen shot of but...
This first pic is with the audio sample rate set as 96000 with acquisition started.
The second is when I stop the acquisition, change the audio rate to 192000 and start the acquisition again.
The third is if I just open the app, add the Funcube and the AM Demod and leave the audio rate at the default of 192000 and then start the acquisition.
![Screenshot 2023-12-05 at 06 30 03](https://github.com/f4exb/sdrangel/assets/25298803/7e99974e-27c8-4184-8b79-546b38629781)
![Screenshot 2023-12-05 at 06 31 11](https://github.com/f4exb/sdrangel/assets/25298803/687fcf7c-b840-4c40-944d-4732eb2cef1d)
![Screenshot 2023-12-05 at 06 37 22](https://github.com/f4exb/sdrangel/assets/25298803/d166bc30-1725-4cf0-80cd-4bbe686d2251)

**Comment by UncleBuck62** [comment]
I have also now attached a sample of the audio I hear.
[20231205 074559.mp3.zip](https://github.com/f4exb/sdrangel/files/13551299/20231205.074559.mp3.zip)

One this I did notice (not sure if it is relevant) is that just under the start/stop acquisition button, even though I changed the audio sample rate to 96k, it still shows 192k.

**Comment by UncleBuck62** [comment]
Further to all of the above, I just thought I would let you know that the FUNcube Pro+ is working with Gqrx v2.11.5 with the default settings of 192k sample rate.

**Comment by UncleBuck62** [comment]
Sorry, I only have a Mac.

My wife has an old windows machine that I MAY be able to use (if I can remember how to use windows).

I'll see what I can do. It may take a few days to work it all out.

**Comment by UncleBuck62** [comment]
I have been doing more testing on this issue and have come across some odd behaviour that may (or may not) shed some light on the problem...

If I open SDRangel, add an audio device, set the device to the FUNcube Dongle v2.0, set the SR to 96k and click the start all devices in workspace button, the device shows as running (according to the spectrum).

If I then add an additional device but this time select the FUNcube Dongle Pro+, I can click the start button on that device and it runs at the full sample rate of 192k (once again, according to the spectrum).

At this point I am able to fully control the FCD using the Funcube device.

Up to this point all looks good, but.... If I add a channel to the Funcube (for example a Broadcast FM Demod), the audio is still choppy and the voices still come out sounding sped up (like chipmunks).

I then tried the whole process again but stopped the audio device and the FUNcube device was still working. I closed the audio device and the FUNcube device was still going, but... as soon as I tried to add the Broadcast FM demod), the program crashed. This only happen once and since then I have been able to open the demod but audio is still chipmunks and choppy.

I hope that the above process makes sense.

**Comment by UncleBuck62** [comment]
This is still an issue, I am not sure why GitHub wants to close it.

**Comment by UncleBuck62** [comment]
This is still an ongoing problem so the issue should not be closed just yet.

**Comment by UncleBuck62** [comment]
This is still an ongoing problem so the issue should not be closed just yet.

**Comment by brauhausdc** [comment]
I was able to get it to work for a bit with the .19 release, but after upgrading to the latest .20 release, I’m back to nothing. It seemed like it wanted to be directly connected  - seemed to work better when I used just a USB A to USB C adapter, vs sending it through a hub. But again, back to nothing. How can I turn on logging or debug to see what is or isn’t going on?    --Dan BrownOn May 5, 2024, at 05:07, UncleBuck62 ***@***.***> wrote:﻿
This is still an ongoing problem so the issue should not be closed just yet.
